From 103be53f6002e2eff0dcae3f95e712fca8c54c05 Mon Sep 17 00:00:00 2001
From: Adam Paquette <acpaquette@usgs.gov>
Date: Fri, 15 Aug 2025 11:23:26 -0700
Subject: [PATCH] Fixes Protobuf compatability issues (#217)

* Regenerate protobuf files with protobuf 4 and pin to protobuf 4 in the environment.yml

* Added changelog entry

* Reverted file changes to keep api the same
---
 CHANGELOG.md                              |   3 +
 environment.yml                           |  32 +++---
 plio/io/ControlNetFileHeaderV0005.proto   |  21 ----
 plio/io/ControlNetFileHeaderV0005_pb2.py  |  16 +--
 plio/io/ControlNetFileV0002.proto         | 128 ----------------------
 plio/io/ControlNetFileV0002_pb2.py        |  22 +---
 plio/io/ControlPointFileEntryV0005.proto  | 118 --------------------
 plio/io/ControlPointFileEntryV0005_pb2.py |  20 +---
 8 files changed, 34 insertions(+), 326 deletions(-)
 delete mode 100644 plio/io/ControlNetFileHeaderV0005.proto
 delete mode 100644 plio/io/ControlNetFileV0002.proto
 delete mode 100644 plio/io/ControlPointFileEntryV0005.proto

diff --git a/CHANGELOG.md b/CHANGELOG.md
index 76a0c9f..383fbf3 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -41,6 +41,9 @@ release.
 - Fixed a bug in which prefixes and suffixes were not properly prepended/appended to point_id [#213](https://github.com/DOI-USGS/plio/issues/213)
 - Fixed a deprecation warning by changing to importlib tools for version determination [#218](https://github.com/DOI-USGS/plio/issues/218)
 
+### Changed
+- Changed protobuf dependency to 4.Y.Z and regenerated protobuf files [#217](https://github.com/DOI-USGS/plio/pull/217)
+
 # [1.6.0]()
 
 ### Added
diff --git a/environment.yml b/environment.yml
index c05a75e..0702bfe 100644
--- a/environment.yml
+++ b/environment.yml
@@ -2,20 +2,22 @@ name: plio
 channels:
   - conda-forge
 dependencies:
-  - gdal 
-  - numpy 
-  - pyproj 
-  - h5py 
-  - pvl >= 1.3.0
-  - scipy 
-  - protobuf 
-  - affine 
-  - networkx 
-  - pandas 
-  - sqlalchemy 
-  - pyyaml 
-  - pytest 
-  - pytest-cov 
-  - coveralls 
+  - affine
+  - coveralls
+  - gdal
+  - h5py
+  - libprotobuf >=4,<5
   - nbsphinx
+  - networkx
+  - numpy
+  - pandas
+  - protobuf >=4,<5
+  - pvl >= 1.3.0
+  - pyproj
+  - pytest
+  - pytest-cov
+  - pyyaml
+  - scipy
+  - setuptools
   - sphinx
+  - sqlalchemy
diff --git a/plio/io/ControlNetFileHeaderV0005.proto b/plio/io/ControlNetFileHeaderV0005.proto
deleted file mode 100644
index ffb1984..0000000
--- a/plio/io/ControlNetFileHeaderV0005.proto
+++ /dev/null
@@ -1,21 +0,0 @@
-// Protocol buffer header descriptor for Isis Control Networks
-//
-// 2018-06-13 Jesse Mapel - Added flag to indicate it is a protobuf 2 file
-// 2024-11-07 Jacob Cain  - Copied into Plio from ISIS3/src/control/objs/ControlNetVersioner
-//                          Generated new _pb2.py with protoc 28.2
-
-syntax="proto2";
-
-package Isis;
-
-message ControlNetFileHeaderV0005 {
-//  Some fields of ControlNet
-  required string networkId    = 1;
-  required string targetName   = 2;
-  optional string created      = 3;
-  optional string lastModified = 4;
-  optional string description  = 5;
-  optional string userName     = 6;
-  optional int32  numPoints    = 7;
-  repeated double targetRadii = 10;
-}
diff --git a/plio/io/ControlNetFileHeaderV0005_pb2.py b/plio/io/ControlNetFileHeaderV0005_pb2.py
index f8b6389..48a71a1 100644
--- a/plio/io/ControlNetFileHeaderV0005_pb2.py
+++ b/plio/io/ControlNetFileHeaderV0005_pb2.py
@@ -1,22 +1,12 @@
 # -*- coding: utf-8 -*-
 # Generated by the protocol buffer compiler.  DO NOT EDIT!
-# NO CHECKED-IN PROTOBUF GENCODE
 # source: ControlNetFileHeaderV0005.proto
-# Protobuf Python Version: 5.28.2
+# Protobuf Python Version: 4.25.3
 """Generated protocol buffer code."""
 from google.protobuf import descriptor as _descriptor
 from google.protobuf import descriptor_pool as _descriptor_pool
-from google.protobuf import runtime_version as _runtime_version
 from google.protobuf import symbol_database as _symbol_database
 from google.protobuf.internal import builder as _builder
-_runtime_version.ValidateProtobufRuntimeVersion(
-    _runtime_version.Domain.PUBLIC,
-    5,
-    28,
-    2,
-    '',
-    'ControlNetFileHeaderV0005.proto'
-)
 # @@protoc_insertion_point(imports)
 
 _sym_db = _symbol_database.Default()
@@ -29,8 +19,8 @@ DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\x1f\x43ontrolNetF
 _globals = globals()
 _builder.BuildMessageAndEnumDescriptors(DESCRIPTOR, _globals)
 _builder.BuildTopDescriptorsAndMessages(DESCRIPTOR, 'ControlNetFileHeaderV0005_pb2', _globals)
-if not _descriptor._USE_C_DESCRIPTORS:
-  DESCRIPTOR._loaded_options = None
+if _descriptor._USE_C_DESCRIPTORS == False:
+  DESCRIPTOR._options = None
   _globals['_CONTROLNETFILEHEADERV0005']._serialized_start=42
   _globals['_CONTROLNETFILEHEADERV0005']._serialized_end=226
 # @@protoc_insertion_point(module_scope)
diff --git a/plio/io/ControlNetFileV0002.proto b/plio/io/ControlNetFileV0002.proto
deleted file mode 100644
index 560a010..0000000
--- a/plio/io/ControlNetFileV0002.proto
+++ /dev/null
@@ -1,128 +0,0 @@
-// 2024-11-07 Jacob Cain - Created by combining ControlNetFileHeaderV0002.proto
-//                         and ControlPointFileEntryV0002.proto from
-//                         ISIS3/src/control/objs/ControlNetVersioner
-//                         Generated new _pb2.py with protoc 28.2
-
-syntax="proto2";
-
-package Isis;
-
-message ControlNetFileHeaderV0002 {
-  //  Some fields of ControlNet
-  required string networkId    = 1;
-  required string targetName   = 2;
-  optional string created      = 3;
-  optional string lastModified = 4;
-  optional string description  = 5;
-  optional string userName     = 6;
-  repeated int32  pointMessageSizes = 7 [packed = true];
-}
-
-message ControlPointFileEntryV0002 {
-  enum PointType {
-    Free                 = 2; //  Image to Image
-    Constrained          = 3; //  Image to Ground
-    Fixed                = 4; //  Image to Ground
-    obsolete_Tie         = 0; //  Image to Image
-    obsolete_Ground      = 1; //  Image to Ground
-  }
-
-  enum AprioriSource {
-    None              = 0;
-    User              = 1;
-    AverageOfMeasures = 2;
-    Reference         = 3;
-    Ellipsoid         = 4;
-    DEM               = 5;
-    Basemap           = 6;
-    BundleSolution    = 7;
-  }
-
-  required string         id   = 1;
-  required PointType      type = 2;
-  optional string         chooserName = 3;
-  optional string         datetime = 4;
-  optional bool           editLock = 5;
-  optional bool           ignore   = 6;
-  optional bool           jigsawRejected = 7;
-  optional int32          referenceIndex = 8;
-
-  optional AprioriSource        aprioriSurfPointSource = 9;
-  optional string               aprioriSurfPointSourceFile = 10;
-  optional AprioriSource        aprioriRadiusSource = 11;
-  optional string               aprioriRadiusSourceFile = 12;
-  optional bool                 latitudeConstrained = 13;
-  optional bool                 longitudeConstrained = 14;
-  optional bool                 radiusConstrained = 15;
-  optional double               aprioriX = 16; // <meters>
-  optional double               aprioriY = 17; // <meters>
-  optional double               aprioriZ = 18; // <meters>
-
-  //  Accuracy of apriori coordinates from basemap, user or mission SPICE,
-  //    jigsaw weights based on these values. The sigmas are stored as a
-  //    3 x 3 variance-covariance matrix.
-  repeated double aprioriCovar = 19 [packed = true];
-
-  optional double adjustedX = 20;
-  optional double adjustedY = 21;
-  optional double adjustedZ = 22;
-
-  //  Set in jigsaw.  Accuracy of adjusted lat/lon from basemap, user or
-  //  mission SPICE stored as a 3 x 3 variance-covariance matrix.
-  repeated double adjustedCovar = 23 [packed = true];
-
-  message PointLogData {
-    optional int32   doubleDataType  = 1;
-    optional double  doubleDataValue = 2;
-    optional int32   boolDataType    = 3;
-    optional bool    boolDataValue   = 4;
-  }
-
-  repeated PointLogData log = 24;
-
-  // Start of ControlMeasure definition
-  message Measure {
-    enum MeasureType {
-      // predicted, unmeasured, unverified
-      Candidate           = 0;
-      // Hand Measured (e.g., qnet)
-      Manual              = 1;
-      // Automatically registered to whole pixel (e.g.,pointreg/qnet)
-      RegisteredPixel     = 2;
-      // Automatically registered to sub-pixel (e.g., pointreg/qnet)
-      RegisteredSubPixel  = 3;
-    }
-
-    message MeasureLogData {
-      optional int32   doubleDataType  = 1;
-      optional double  doubleDataValue = 2;
-      optional int32   boolDataType    = 3;
-      optional bool    boolDataValue   = 4;
-    }
-
-    required string       serialnumber   = 1;
-    required MeasureType  type           = 2;
-    optional double       sample         = 3;
-    optional double       line           = 4;
-    // Jigsaw info-solution error <pixels>
-    optional double       sampleResidual = 5;
-    optional double       lineResidual   = 6;
-    optional string       choosername    = 7;
-    optional string       datetime       = 8;
-    optional bool         editLock       = 9;
-    optional bool         ignore         = 10;
-    optional bool         jigsawRejected = 11;
-    optional double       diameter       = 12;
-    // The first identified location of measure
-    optional double       apriorisample  = 13;
-    optional double       aprioriline    = 14;
-
-    // Uncertainty/Sigma of current samp/line <pixels>
-    optional double       samplesigma    = 15;
-    optional double       linesigma      = 16;
-    repeated MeasureLogData log          = 17;
-  }
-
-  //  Actual declaration of variable number of ControlMeasures
-  repeated Measure measures = 25;
-}
diff --git a/plio/io/ControlNetFileV0002_pb2.py b/plio/io/ControlNetFileV0002_pb2.py
index 89f23d7..0d674d0 100644
--- a/plio/io/ControlNetFileV0002_pb2.py
+++ b/plio/io/ControlNetFileV0002_pb2.py
@@ -1,22 +1,12 @@
 # -*- coding: utf-8 -*-
 # Generated by the protocol buffer compiler.  DO NOT EDIT!
-# NO CHECKED-IN PROTOBUF GENCODE
 # source: ControlNetFileV0002.proto
-# Protobuf Python Version: 5.28.2
+# Protobuf Python Version: 4.25.3
 """Generated protocol buffer code."""
 from google.protobuf import descriptor as _descriptor
 from google.protobuf import descriptor_pool as _descriptor_pool
-from google.protobuf import runtime_version as _runtime_version
 from google.protobuf import symbol_database as _symbol_database
 from google.protobuf.internal import builder as _builder
-_runtime_version.ValidateProtobufRuntimeVersion(
-    _runtime_version.Domain.PUBLIC,
-    5,
-    28,
-    2,
-    '',
-    'ControlNetFileV0002.proto'
-)
 # @@protoc_insertion_point(imports)
 
 _sym_db = _symbol_database.Default()
@@ -29,13 +19,13 @@ DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\x19\x43ontrolNetF
 _globals = globals()
 _builder.BuildMessageAndEnumDescriptors(DESCRIPTOR, _globals)
 _builder.BuildTopDescriptorsAndMessages(DESCRIPTOR, 'ControlNetFileV0002_pb2', _globals)
-if not _descriptor._USE_C_DESCRIPTORS:
-  DESCRIPTOR._loaded_options = None
-  _globals['_CONTROLNETFILEHEADERV0002'].fields_by_name['pointMessageSizes']._loaded_options = None
+if _descriptor._USE_C_DESCRIPTORS == False:
+  DESCRIPTOR._options = None
+  _globals['_CONTROLNETFILEHEADERV0002'].fields_by_name['pointMessageSizes']._options = None
   _globals['_CONTROLNETFILEHEADERV0002'].fields_by_name['pointMessageSizes']._serialized_options = b'\020\001'
-  _globals['_CONTROLPOINTFILEENTRYV0002'].fields_by_name['aprioriCovar']._loaded_options = None
+  _globals['_CONTROLPOINTFILEENTRYV0002'].fields_by_name['aprioriCovar']._options = None
   _globals['_CONTROLPOINTFILEENTRYV0002'].fields_by_name['aprioriCovar']._serialized_options = b'\020\001'
-  _globals['_CONTROLPOINTFILEENTRYV0002'].fields_by_name['adjustedCovar']._loaded_options = None
+  _globals['_CONTROLPOINTFILEENTRYV0002'].fields_by_name['adjustedCovar']._options = None
   _globals['_CONTROLPOINTFILEENTRYV0002'].fields_by_name['adjustedCovar']._serialized_options = b'\020\001'
   _globals['_CONTROLNETFILEHEADERV0002']._serialized_start=36
   _globals['_CONTROLNETFILEHEADERV0002']._serialized_end=211
diff --git a/plio/io/ControlPointFileEntryV0005.proto b/plio/io/ControlPointFileEntryV0005.proto
deleted file mode 100644
index c26f6aa..0000000
--- a/plio/io/ControlPointFileEntryV0005.proto
+++ /dev/null
@@ -1,118 +0,0 @@
-// Protocol buffer control point descriptor for Isis Control Networks
-//
-// 2018-06-13 Jesse Mapel - Added flag to indicate it is a protobuf 2 file
-// 2024-11-07 Jacob Cain  - Copied into Plio from ISIS3/src/control/objs/ControlNetVersioner
-//                          Generated new _pb2.py with protoc 28.2
-
-syntax="proto2";
-
-package Isis;
-
-message ControlPointFileEntryV0005 {
-  enum PointType {
-    Free                 = 2; //  Image to Image
-    Constrained          = 3; //  Image to Ground
-    Fixed                = 4; //  Image to Ground
-    obsolete_Tie         = 0; //  Image to Image
-    obsolete_Ground      = 1; //  Image to Ground
-  }
-
-  enum AprioriSource {
-    None              = 0;
-    User              = 1;
-    AverageOfMeasures = 2;
-    Reference         = 3;
-    Ellipsoid         = 4;
-    DEM               = 5;
-    Basemap           = 6;
-    BundleSolution    = 7;
-  }
-
-  required string         id   = 1;
-  required PointType      type = 2;
-  optional string         chooserName = 3;
-  optional string         datetime = 4;
-  optional bool           editLock = 5;
-  optional bool           ignore   = 6;
-  optional bool           jigsawRejected = 7;
-  optional int32          referenceIndex = 8;
-
-  optional AprioriSource        aprioriSurfPointSource = 9;
-  optional string               aprioriSurfPointSourceFile = 10;
-  optional AprioriSource        aprioriRadiusSource = 11;
-  optional string               aprioriRadiusSourceFile = 12;
-  optional bool                 latitudeConstrained = 13;
-  optional bool                 longitudeConstrained = 14;
-  optional bool                 radiusConstrained = 15;
-  optional double               aprioriX = 16; // <meters>
-  optional double               aprioriY = 17; // <meters>
-  optional double               aprioriZ = 18; // <meters>
-
-  //  Accuracy of apriori coordinates from basemap, user or mission SPICE,
-  //    jigsaw weights based on these values. The sigmas are stored as a
-  //    3 x 3 variance-covariance matrix.
-  repeated double aprioriCovar = 19 [packed = true];
-
-  optional double adjustedX = 20;
-  optional double adjustedY = 21;
-  optional double adjustedZ = 22;
-
-  //  Set in jigsaw.  Accuracy of adjusted lat/lon from basemap, user or
-  //  mission SPICE stored as a 3 x 3 variance-covariance matrix.
-  repeated double adjustedCovar = 23 [packed = true];
-
-  message PointLogData {
-    optional int32   doubleDataType  = 1;
-    optional double  doubleDataValue = 2;
-    optional int32   boolDataType    = 3;
-    optional bool    boolDataValue   = 4;
-  }
-
-  repeated PointLogData log = 24;
-
-// Start of ControlMeasure definition
-  message Measure {
-    enum MeasureType {
-      // predicted, unmeasured, unverified
-      Candidate           = 0;
-      // Hand Measured (e.g., qnet)
-      Manual              = 1;
-      // Automatically registered to whole pixel (e.g.,pointreg/qnet)
-      RegisteredPixel     = 2;
-      // Automatically registered to sub-pixel (e.g., pointreg/qnet)
-      RegisteredSubPixel  = 3;
-    }
-
-    message MeasureLogData {
-      optional int32   doubleDataType  = 1;
-      optional double  doubleDataValue = 2;
-      optional int32   boolDataType    = 3;
-      optional bool    boolDataValue   = 4;
-    }
-
-    required string       serialnumber   = 1;
-    required MeasureType  type           = 2;
-    optional double       sample         = 3;
-    optional double       line           = 4;
-    // Jigsaw info-solution error <pixels>
-    optional double       sampleResidual = 5;
-    optional double       lineResidual   = 6;
-    optional string       choosername    = 7;
-    optional string       datetime       = 8;
-    optional bool         editLock       = 9;
-    optional bool         ignore         = 10;
-    optional bool         jigsawRejected = 11;
-    optional double       diameter       = 12;
-    // The first identified location of measure
-    optional double       apriorisample  = 13;
-    optional double       aprioriline    = 14;
-
-    // Uncertainty/Sigma of current samp/line <pixels>
-    optional double       samplesigma    = 15;
-    optional double       linesigma      = 16;
-    repeated MeasureLogData log          = 17;
-  }
-
-  //  Actual declaration of variable number of ControlMeasures
-  repeated Measure measures = 25;
-}
diff --git a/plio/io/ControlPointFileEntryV0005_pb2.py b/plio/io/ControlPointFileEntryV0005_pb2.py
index 5776b73..126b72f 100644
--- a/plio/io/ControlPointFileEntryV0005_pb2.py
+++ b/plio/io/ControlPointFileEntryV0005_pb2.py
@@ -1,22 +1,12 @@
 # -*- coding: utf-8 -*-
 # Generated by the protocol buffer compiler.  DO NOT EDIT!
-# NO CHECKED-IN PROTOBUF GENCODE
 # source: ControlPointFileEntryV0005.proto
-# Protobuf Python Version: 5.28.2
+# Protobuf Python Version: 4.25.3
 """Generated protocol buffer code."""
 from google.protobuf import descriptor as _descriptor
 from google.protobuf import descriptor_pool as _descriptor_pool
-from google.protobuf import runtime_version as _runtime_version
 from google.protobuf import symbol_database as _symbol_database
 from google.protobuf.internal import builder as _builder
-_runtime_version.ValidateProtobufRuntimeVersion(
-    _runtime_version.Domain.PUBLIC,
-    5,
-    28,
-    2,
-    '',
-    'ControlPointFileEntryV0005.proto'
-)
 # @@protoc_insertion_point(imports)
 
 _sym_db = _symbol_database.Default()
@@ -29,11 +19,11 @@ DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n ControlPointFileE
 _globals = globals()
 _builder.BuildMessageAndEnumDescriptors(DESCRIPTOR, _globals)
 _builder.BuildTopDescriptorsAndMessages(DESCRIPTOR, 'ControlPointFileEntryV0005_pb2', _globals)
-if not _descriptor._USE_C_DESCRIPTORS:
-  DESCRIPTOR._loaded_options = None
-  _globals['_CONTROLPOINTFILEENTRYV0005'].fields_by_name['aprioriCovar']._loaded_options = None
+if _descriptor._USE_C_DESCRIPTORS == False:
+  DESCRIPTOR._options = None
+  _globals['_CONTROLPOINTFILEENTRYV0005'].fields_by_name['aprioriCovar']._options = None
   _globals['_CONTROLPOINTFILEENTRYV0005'].fields_by_name['aprioriCovar']._serialized_options = b'\020\001'
-  _globals['_CONTROLPOINTFILEENTRYV0005'].fields_by_name['adjustedCovar']._loaded_options = None
+  _globals['_CONTROLPOINTFILEENTRYV0005'].fields_by_name['adjustedCovar']._options = None
   _globals['_CONTROLPOINTFILEENTRYV0005'].fields_by_name['adjustedCovar']._serialized_options = b'\020\001'
   _globals['_CONTROLPOINTFILEENTRYV0005']._serialized_start=43
   _globals['_CONTROLPOINTFILEENTRYV0005']._serialized_end=1837
-- 
2.39.5 (Apple Git-154)

